cargo-spatial/src/lib.rs

```diff
@@ -0,0 +1,23 @@
+pub mod codegen;
+pub mod config;
+pub mod local;
+pub mod opt;
+
+/// Generates a random, valid component ID.
```

Comment:
I'm not sure about the usefulness of this command. Generally, you would arrange your schema so that the ids have some semantic meaning (or at least I would). Such that grouped schema ids = grouped features.

Reply:
So that's something that I wanted to ask more about. In relation to #56 and the desire to be able to distribute schema files via crates.io (or really any package manager), there needs to be a way to guarantee that your component IDs are unique not just within your own schema files, but also with any component definitions that you pull in from a third party. Generally, the way this is handled is by generating random IDs and having the range of valid values be so huge as to make collision a statistical impossibility (i.e. UUIDs or Git commit hashes). I added the random generation functionality in order to assist this approach, though I worry that the fact that component IDs are limited to the range 100 to 536,870,911 (with a range of 10,000 values taken out of the middle) means that collisions are going to be still pretty likely.

How is this issue handled for other language SDKs? Do other languages even support pulling in schema definition files from a package manager?

Reply:
Its generally an adhoc thing - the GDKs are the first to really play with the idea of shipping schema as user code and we are still figuring it out.